### ISSUE TITLE
@Transactional with autoCommit=true, similar to readOnly=true

### ISSUE THREAD

**Issue by jcranfordupgrade** [spring-projects/spring-boot#40269]
I would like to use one DataSource bean for a mix of these operations:
1. readOnly=true (implies autoCommit=true)
2. readOnly=false with autoCommit=false
3. readOnly=false with autoCommit=true

One option might be @Transactional annotation supporting `autoCommit` and `readOnly` tags; it only has `readOnly` tag today.

Another option might be two out-of-the-box `PlatformTransactionManager` beans; where @Transactional `value` tag can be used pick DataSourceTransactionManager (aka AutoCommitOffTransactionManager) versus AutoCommitOnTransactionManager (new).

I am open to other solutions too, these are just suggestions.

In general, autoCommit on or off is a convenience feature. Both of these examples are transactions.
```
SET AUTOCOMMIT=0;
BEGIN;
UPDATE table SET version=version+1 WHERE version=0;
COMMIT;
```
```
SET AUTOCOMMIT=1;
UPDATE table SET version=version+1 WHERE version=0;
```

Both examples are transactional, because all of the affected rows are updated atomically. If a connection pool parks connections with AUTOCOMMIT=1, then the 2nd example only requires one round-trip to the database.

1. For high latency connections, using autoCommit=false does 4 expensive round-trips, while using autoCommit=true only does 1 expensive round-trip to execute the one-statement UPDATE. Both are transactional, but one is much faster.
2. For using @Version in READ COMMITTED mode, the first transaction requires application code to use OptimisticLockingException and Retryable due to competing version column updates. For the second transaction example, the DB would handle optimistic locking much more efficiently, and Retryable may not be needed at all.

I opened a Stack Exchange question, but the answers I got were not helpful.

https://stackoverflow.com/questions/78268584/how-to-mix-autocommit-true-and-autocommit-true-transactional-operations-in

In summary, I would like to see Spring support using a single DataSource bean with a mix of autoCommit=true and autoCommit=false in user applications. Setting autoCommit=false at the DataSource level doesn't work, because I would need to create two DataSource beans, and I would end up with separate connection pools.
